sql/core/src/main/scala/org/apache/spark/sql/execution/SparkPlanner.scala

```diff
@@ -44,8 +44,7 @@ class SparkPlanner(val sqlContext: SQLContext) extends SparkStrategies {
       EquiJoinSelection ::
       InMemoryScans ::
       BasicOperators ::
-      CartesianProduct ::
-      BroadcastNestedLoopJoin :: Nil)
+      CartesianProduct :: Nil)
```

Comment:
I am not sure if `BroadcastNestedLoopJoin` is a good last gate. It is possible that we join two large tables and we cannot really use broadcast join, right?